### ISSUE TITLE
(XMB) Problem on Disk Control / Change Disk

### ISSUE THREAD

**Issue by LuanZorde** [libretro/RetroArch#7365]
Hi, a problem in the Disk Control. When apend a image of New Disk he open a New Disk but dont load a save game.
 Please correct this error

**Comment by LuanZorde** [comment]
This on Retroarch on Android

**Comment by fr500** [comment]
Please use the template the next time. It's there for a reason.

The issue is on RetroArch side.
This also affects subsystem

The problem:

When trying to append a disk RetroArch won't append the disk but load as new content instead.
This also happens when loading subsystem type content (bsnes / sameboy / tgb dual / fba).

This happens because of a bug in the RetroArch file browser code. Whenever you reach the "drives list" (the topmost directory in the file list where it would show C:\ D:\ or  / and /sdcard on android or whatever the platform has) the action and the file extension filter is reset.

To work around this problem you can just set a directory under Directory / File Browser

**Comment by fr500** [comment]
Basically what happens is:

![image](https://user-images.githubusercontent.com/1721040/46511337-b3b94f00-c813-11e8-8016-07e5ecb03d53.png)

![image](https://user-images.githubusercontent.com/1721040/46511342-b6b43f80-c813-11e8-8a30-36c347b401e2.png)

After that "folder" is shown the extension filter is reset (and the file selection action is reset too)

For instance it shows gcz files when adding a disk for psx when gcz is not in the supported extension list
![image](https://user-images.githubusercontent.com/1721040/46511359-d6e3fe80-c813-11e8-89b7-9dd01212485f.png)

---

If you apply the workaround and navigate to the same gcn folder when trying to append a disk:

![image](https://user-images.githubusercontent.com/1721040/46511396-0bf05100-c814-11e8-9d0b-119d53ee2dad.png)

Which is the correct behavior.

Basically getting to the drives list breaks the extension filter and the select file action

**Comment by LuanZorde** [comment]
So, I using the Retroarch for Android...

Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by fr500** [comment]
I doesn't matter the code is identical

**Comment by LuanZorde** [comment]
Ok friend.
So I select a Directory?
________________________________

**Comment by LuanZorde** [comment]
[Imagem]

This isso what apears tô me...


Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by LuanZorde** [comment]
When I apend a Disk

Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by fr500** [comment]
I know the issue I was explaining it better so the dev can fix it

**Comment by LuanZorde** [comment]
Ok, I wait the correction of this issue...

Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by fr500** [comment]
The problem is not only on disk image append, ie: it's not specific to disk image append.
It can happen on any instance on the filebrowser if you get to the top level directory

With top level directory I mean this

![image](https://user-images.githubusercontent.com/1721040/47277487-74fef500-d586-11e8-95c3-77dcaa354c7d.png)

If you try to load an overlay and get to that, it will try to **load the overlay as content**
If you try to load a shader and get to that list, it will try to **load the shader as content**

Etc.

**Comment by fr500** [comment]
Seems it can't be triggered with overlays or shaders because another bug, doing "Parent Directory" in "D:\" leads to the RetroArch dir 🤣 (which means you could never load overlays in drive C: or E: or whatever I guess which is not a problem, but if a content dir is set you would never be able to load roms in another disk, a CD, a USB drive or a different mountpoint on linux, etc, anyway that's another bug)

**Comment by orbea** [comment]
Pretty sure this is a duplicate of issue https://github.com/libretro/RetroArch/issues/5642.
